File: C#/디자인패턴/Iterator/Array.cs
public class Array : Aggregator {
    private Item[] items;
    public Array(Item[] items) {
        this.items = items;
    }
    public Item gItem(int idx) {
        return items[idx];
    }
    public int getCount() {
        return items.Length;
    }

    public Iterator iterator()
    {
        return new ArrayIterator(this);
    }
}

File: Flutter/디자이패턴(dart.ver)/Iterator/Array.dart
import "ArrayIterator.dart";
import "Aggreator.dart";
import "M_Item.dart";

class Array extends Aggreator {
  List<M_Item> m_items = [];

  Array(List<M_Item> items_list) {
    this.m_items = items_list;
  }

  M_Item getItem(int idx) {
    return this.m_items[idx];
  }

  int getCount() {
    return this.m_items.length;
  }

  ArrayIterator iterator() {
    return ArrayIterator(this);
  }
}